src/app/shared/change-password-dialog/change-password-dialog.component.ts

```diff
@@ -0,0 +1,43 @@
+/** Angular Imports */
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+import { FormBuilder, Validators } from '@angular/forms';
+
+/** Custom Services */
+import { PasswordService } from './password.service';
+
+/**
+ * Change Password Dialog component.
+ */
+@Component({
+  selector: 'mifosx-change-password-dialog',
+  templateUrl: './change-password-dialog.component.html',
+  styleUrls: ['./change-password-dialog.component.scss']
+})
+export class ChangePasswordDialogComponent implements OnInit {
+
+  /** Change Password Form */
+  changePasswordForm: any;
+
+  /**
+   * @param {MatDialogRef} dialogRef Component reference to dialog.
+   * @param {any} data Provides any data.
+   */
+  constructor(public dialogRef: MatDialogRef<ChangePasswordDialogComponent>,
+              @Inject(MAT_DIALOG_DATA) public data: any,
+              private passwordCheckService: PasswordService,
+              private formBuilder: FormBuilder) { }
+
+  ngOnInit() {
+    this.createChangePasswordForm();
+  }
+
+  /** Change Password form */
+  createChangePasswordForm() {
+    this.changePasswordForm = this.formBuilder.group({
+      'password': ['', [Validators.required, Validators.pattern('^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*[$@$!%*?&])[A-Za-z\d$@$!%*?&].{8,}$'), Validators.maxLength(50), Validators.minLength(8)]],
+      'repeatPassword': ['', [Validators.required, this.passwordCheckService.confirmPassword('password')]]
```

Comment:
password service

Reply:
@BLasan Muskan will guide you through this

Reply:
Ok. Any errors encountered?

Reply:
it's not passwordcheckservice it's password service

Reply:
I moved the file to component.ts . Now there's no service file